src/js/webrequest.js

```diff
@@ -325,6 +325,65 @@ function blockMozCspReports(details) {
   }
 }
 
+/**
+ * A backstop to remove undesirable redirects for when our JS approach fails.
+ *
+ * Some Google properties (such as Google Docs in Firefox) rewrite clicked
+ * link actions via some mix of capturing all events, window.open() with
+ * document.write(), and meta refresh.
+ *
+ * @param {Object} details webRequest request details object
+ *
+ * @returns {Object|undefined} Can redirect requests
+ */
+function bypassGoogleRedirects(details) {
+  if (!badger.INITIALIZED) { return; }
+  let request_host = extractHostFromURL(details.url);
+  if (!badger.isPrivacyBadgerEnabled(request_host)) { return; }
+  let urlObj, redirect_url;
+  try {
+    urlObj = new URL(details.url);
+  } catch (e) { /* ignore */ }
+  if (urlObj && urlObj.searchParams) {
+    redirect_url = urlObj.searchParams.get('url') || urlObj.searchParams.get('q');
+  }
+  if (redirect_url) {
+    if (redirect_url.startsWith("https://") || redirect_url.startsWith("http://")) {
+      if (utils.isThirdPartyDomain(extractHostFromURL(redirect_url), request_host)) {
+        return {
+          redirectUrl: redirect_url
+        };
+      }
+    }
+  }
```

Comment:
Suggesting some logic eval set to variables so to avoid the nested ifs.
```suggestion
  let isHttpUrl = redirect_url.startsWith('https://') || redirect_url.startsWith('http://');
  let isThirdPartyDomain = utils.isThirdPartyDomain(extractHostFromURL(redirect_url), request_host);

  if (redirect_url && isHttpUrl && isThirdPartyDomain ) {
    return {
      redirectUrl: redirect_url
    };
  }
```

Reply:
If we don't have a redirect URL or it's not a valid URL, we don't need to do the work of host extraction and third party checking. I agree though that the nesting is not great and if it gets any messier, it should be refactored.